src/ui/connect/connector.ts

```diff
@@ -35,73 +23,80 @@ export default class Connector implements ExtensionActions {
+        }
     }
 
     enable() {
-        this.port.postMessage({type: 'enable'});
+        chrome.runtime.sendMessage({name: 'ui', type: 'enable'});
     }
 
     disable() {
-        this.port.postMessage({type: 'disable'});
+        chrome.runtime.sendMessage({name: 'ui', type: 'disable'});
     }
 
     setShortcut(command: string, shortcut: string) {
-        this.port.postMessage({type: 'set-shortcut', data: {command, shortcut}});
+        chrome.runtime.sendMessage({name: 'ui', type: 'set-shortcut', data: {command, shortcut}});
     }
 
     changeSettings(settings: Partial<UserSettings>) {
-        this.port.postMessage({type: 'change-settings', data: settings});
+        chrome.runtime.sendMessage({name: 'ui', type: 'change-settings', data: settings});
     }
 
     setTheme(theme: Partial<FilterConfig>) {
-        this.port.postMessage({type: 'set-theme', data: theme});
+        chrome.runtime.sendMessage({name: 'ui', type: 'set-theme', data: theme});
     }
 
     toggleURL(url: string) {
-        this.port.postMessage({type: 'toggle-url', data: url});
+        chrome.runtime.sendMessage({name: 'ui', type: 'toggle-url', data: url});
     }
 
     markNewsAsRead(ids: string[]) {
-        this.port.postMessage({type: 'mark-news-as-read', data: ids});
+        chrome.runtime.sendMessage({name: 'ui', type: 'mark-news-as-read', data: ids});
     }
 
     loadConfig(options: {local: boolean}) {
-        this.port.postMessage({type: 'load-config', data: options});
+        chrome.runtime.sendMessage({name: 'ui', type: 'load-config', data: options});
     }
 
     async applyDevDynamicThemeFixes(text: string) {
         return await this.sendRequest<void>({type: 'apply-dev-dynamic-theme-fixes', data: text}, ({error}, resolve, reject) => error ? reject(error) : resolve());
     }
 
     resetDevDynamicThemeFixes() {
-        this.port.postMessage({type: 'reset-dev-dynamic-theme-fixes'});
+        chrome.runtime.sendMessage({name: 'ui', type: 'reset-dev-dynamic-theme-fixes'});
     }
 
     async applyDevInversionFixes(text: string) {
         return await this.sendRequest<void>({type: 'apply-dev-inversion-fixes', data: text}, ({error}, resolve, reject) => error ? reject(error) : resolve());
     }
 
     resetDevInversionFixes() {
-        this.port.postMessage({type: 'reset-dev-inversion-fixes'});
+        chrome.runtime.sendMessage({name: 'ui', type: 'reset-dev-inversion-fixes'});
     }
 
     async applyDevStaticThemes(text: string) {
         return await this.sendRequest<void>({type: 'apply-dev-static-themes', data: text}, ({error}, resolve, reject) => error ? reject(error) : resolve());
     }
 
     resetDevStaticThemes() {
-        this.port.postMessage({type: 'reset-dev-static-themes'});
+        chrome.runtime.sendMessage({name: 'ui', type: 'reset-dev-static-themes'});
     }
 
     disconnect() {
-        this.port.disconnect();
+        if (this.changeSubscribers.size > 0) {
+            this.changeSubscribers = new Set();
```

Comment:
```suggestion
            this.changeSubscribers.clear();
```
It's better to clear a Set than creating a new one.